src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,102 @@
+package duke;
+
+import java.util.ArrayList;
+
+import duke.enums.Views;
+import duke.task.Task;
+
+/**
+ * TaskList object to store the list of tasks
+ */
+public class TaskList {
+    private ArrayList<Task> tasksList;
+
+    /**
+     * For creating a taskList from storage
+     *
+     * @param tasksList as an ArrayList
+     * @throws DukeException
+     */
+    TaskList(ArrayList<Task> tasksList) throws DukeException {
+        this.tasksList = tasksList;
+    }
+
+    /**
+     * Creates a empty task list
+     */
+    TaskList() {
+        this.tasksList = new ArrayList<Task>(100);
+    }
+
+    /**
+     * Gets the Task at the index specified
+     *
+     * @param index of the task in the ArrayList
+     * @return Task found from the ArrayList
+     * @throws DukeException
+     */
+    public Task get(int index) throws DukeException {
+        try {
+            return tasksList.get(index);
+        } catch (Exception e) {
+            throw new DukeException(Views.OUT_RANGE_ERR_STRING.eng());
+        }
+    }
+
+    /**
+     * Gets the array list for the list of items
+     *
+     * @return ArrayList of tasks
+     */
+    public ArrayList<Task> getList() {
+        return tasksList;
+    }
+
+    /**
+     * Searches and returns an arraylist for results
+     *
+     * @return ArrayList of found task
+     */
+    public ArrayList<Task> search(String query) {
+        ArrayList<Task> result = new ArrayList<Task>();
+        for (Task task : this.tasksList) {
+            if (task.getTitle().toLowerCase().contains(query.trim().toLowerCase())) {
+                result.add(task);
+            }
+        }
+        return result;
+    }
+
+    /**
+     * @return int size of the taskList
+     */
+    public int size() {
+        return tasksList.size();
+    }
+
+    /**
+     * Adds a new task into the list
+     *
+     * @param newTask new Task object to be added into the list
+     */
+    public void add(Task newTask) {
+        this.tasksList.add(newTask);
+    }
+
+    /**
+     * Removes the task given the index of it
+     *
+     * @param taskNo int index of task in the ArrayList
+     */
+    public void remove(int taskNo) {
```

Comment:
As taskNo might be informal english and could be confused with the word "No", perhaps the variable name taskNo could be changed to taskNumber?

Reply:
Good catch! I'll use taskNum for less keystrokes